code/parachain/frame/assets-registry/src/lib.rs

```diff
@@ -253,30 +347,52 @@ pub mod pallet {
 
 	impl<T: Config> RemoteAssetRegistryMutate for Pallet<T> {
 		type AssetId = T::LocalAssetId;
-
 		type AssetNativeLocation = T::ForeignAssetId;
-
 		type Balance = T::Balance;
 
+		fn register_asset(
+			asset_id: Self::AssetId,
+			location: Option<Self::AssetNativeLocation>,
+			ratio: Option<Rational>,
+			name: Vec<u8>,
+			symbol: Vec<u8>,
+			decimals: u8,
+		) -> DispatchResult {
```

Comment:
New function allowing other protocols to directly register new assets via trait.